crypto/keys/src/ed25519.rs

```diff
@@ -86,18 +98,20 @@ impl DistinguishedEncoding for Ed25519Signature {
     }
 
     /// Serializes this object into a DER-encoded SubjectPublicKeyInfo structure
-    fn to_der(&self) -> Vec<u8> {
+    fn to_der_slice<'a>(&self, buff: &'a mut [u8]) -> &'a [u8] {
```

Comment:
Consider keeping the existing method, making it forward to the slice method

```suggestion
    #[cfg(feature = "alloc")]
    fn to_der(&self) -> Vec<u8> {
      let result = vec![0; self.der_size()];
      self.to_der_slice(&mut result);
      result
    }

    /// Serializes this object into a DER-encoded SubjectPublicKeyInfo structure, in the provided buffer.
    fn to_der_slice<'a>(&self, buff: &'a mut [u8]) -> &'a [u8] {
```

Reply:
the existing method is provided for backwards compatibility as a default in `DistinguishedEncoding::to_der` so it's not required for each impl.

Reply:
The current method is implemented in the trait.